From 53e09eeea61d41a2dbc536090fe746b6fe335886 Mon Sep 17 00:00:00 2001
From: Bharat Bhushan <Bharat.Bhushan@nxp.com>
Date: Mon, 29 Jan 2018 16:57:18 +0800
Subject: [PATCH 540/706] vfio fsl-mc: Enable VFIO read/writes for DPMCP
 objects

MC commands can come over DPMCP objects as well, so we enabled
read/writes on DPMCP objects as well.

Signed-off-by: Bharat Bhushan <Bharat.Bhushan@nxp.com>
(cherry picked from commit ff0307ffadb3e233bdb40de0f176039c1c2c38e9)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/vfio/fsl-mc/vfio_fsl_mc.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/drivers/vfio/fsl-mc/vfio_fsl_mc.c b/drivers/vfio/fsl-mc/vfio_fsl_mc.c
index b997fb2f1069..6a339cf78428 100644
--- a/drivers/vfio/fsl-mc/vfio_fsl_mc.c
+++ b/drivers/vfio/fsl-mc/vfio_fsl_mc.c
@@ -288,8 +288,9 @@ static ssize_t vfio_fsl_mc_read(void *device_data, char __user *buf,
 	uint64_t data[8];
 	int i;
 
-	/* Read ioctl supported only for DPRC device */
-	if (strcmp(vdev->mc_dev->obj_desc.type, "dprc"))
+	/* Read ioctl supported only for DPRC and DPMCP device */
+	if (strcmp(vdev->mc_dev->obj_desc.type, "dprc") &&
+	    strcmp(vdev->mc_dev->obj_desc.type, "dpmcp"))
 		return -EINVAL;
 
 	if (index >= vdev->num_regions)
@@ -390,8 +391,9 @@ static ssize_t vfio_fsl_mc_write(void *device_data, const char __user *buf,
 	uint64_t data[8];
 	int ret;
 
-	/* Write ioctl supported only for DPRC device */
-	if (strcmp(vdev->mc_dev->obj_desc.type, "dprc"))
+	/* Write ioctl supported only for DPRC and DPMCP device */
+	if (strcmp(vdev->mc_dev->obj_desc.type, "dprc") &&
+	    strcmp(vdev->mc_dev->obj_desc.type, "dpmcp"))
 		return -EINVAL;
 
 	if (index >= vdev->num_regions)
-- 
2.17.1

